src/mono/wasm/runtime/driver.c

```diff
@@ -969,6 +969,29 @@ mono_wasm_get_type_aqn (MonoType * typePtr) {
 	return mono_type_get_name_full (typePtr, MONO_TYPE_NAME_FORMAT_ASSEMBLY_QUALIFIED);
 }
 
+// this will return bool value if the object is a bool, otherwise it will return -1 or error
+EMSCRIPTEN_KEEPALIVE int
+mono_wasm_read_value_if_bool_unsafe (PVOLATILE(MonoObject) obj) {
+	MONO_ENTER_GC_UNSAFE;
+
+	MonoClass *klass = mono_object_get_class (obj);
+	if (!klass)
+		return MARSHAL_ERROR_NULL_CLASS_POINTER;
+
+	MonoType *type = mono_class_get_type (klass), *original_type = type;
```

Comment:
nit: `*original_type = type;` in a separate line otherwise it seems easy to miss while reading.

Reply:
oh, I missed that in my copy&paste. Removed.